### PR TITLE
Tags support config aggregator & Bug fix for settings api

### PR DESCRIPTION
**Pull request by Ramya-c4**
GIT ISSUE : 

https://github.ibm.com/devx-app-services/configuration-aggregator-planning/issues/339
https://github.ibm.com/devx-app-services/configuration-aggregator-planning/issues/362

testing output : 

BUG FIX

https://github.ibm.com/devx-app-services/configuration-aggregator-planning/issues/362#issuecomment-96850360

TAGS AND ENTERPRISE ISSUE

```
ibm_config_aggregator_configurations = {
  "configs" = [
    {
      "about" = {
        "access_tag" = [
          "test:stage",
        ]
        "account_id" = "c1d20fee2fe24c42b8ef6583283d2dcf"
        "config_type" = "instance"
        "last_config_refresh_time" = "2024-11-12T00:30:33.000Z"
        "location" = "us-south"
        "resource_crn" = "crn:v1:staging:public:apprapp:us-south:a/c1d20fee2fe24c42b8ef6583283d2dcf:73656baf-8627-4802-8bce-c9db7b94a8aa::"
        "resource_group_id" = "8e823040552444afbcbd702354891fed"
        "resource_name" = "Stage-Enterprise-Test"
        "service_name" = "apprapp"
        "service_tag" = []
        "user_tag" = [
          "stage",
        ]
      }
      "config" = {
        "cbr_check_enforced" = ""
        "cbr_private_public_allowed_ip_list" = null
        "endpoints_restricted" = false
        "restricts_private_endpoints" = false
        "wild_card_in_private_list" = false
        "wild_card_in_public_list" = false
      }
    },
  ]
}
```

### CODE REVIEW

**Comment by Ramya-c4** [comment]
image
image

File: ibm/service/configurationaggregator/data_source_ibm_config_aggregator_configurations.go [hunk 1]
```diff
@@ -27,65 +27,82 @@ func DataSourceIbmConfigAggregatorConfigurations() *schema.Resource {
...
+			},
+			"user_tags": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Filter the resource configurations attached with the specified user tags.",
+			},
+			"service_tags": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Filter the resource configurations attached with the specified access tags.",
```
Comment (VaishnaviGopal) [inline ibm/service/configurationaggregator/data_source_ibm_config_aggregator_configurations.go:73]: access tags and service tags have same description. can you look into it?

Comment (Ramya-c4) [reply ibm/service/configurationaggregator/data_source_ibm_config_aggregator_configurations.go]: addressed

File: ibm/service/configurationaggregator/data_source_ibm_config_aggregator_configurations.go [hunk 2]
```diff
@@ -27,65 +27,82 @@ func DataSourceIbmConfigAggregatorConfigurations() *schema.Resource {
...
-			"resource_crn": &schema.Schema{
+			"resource_crn": {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Description: "The crn of the resource.",
 			},
-			"prev": &schema.Schema{
+			"sub_account": {
+				Type:        schema.TypeString,
+				Optional:    true,
```
Comment (hkantare) [inline ibm/service/configurationaggregator/data_source_ibm_config_aggregator_configurations.go:57]: Add Validation function to check the condition described in docs
```
The maximum length is `128` characters. The minimum length is `1` character. The value must match regular expression `[a-zA-Z0-9]`
```

Comment (Ramya-c4) [reply ibm/service/configurationaggregator/data_source_ibm_config_aggregator_configurations.go]: image
image
image
image

**Comment by Ramya-c4** [comment]
image
image
image
image

**Review by hkantare — APPROVED** [review]

**Pull request merged**